src/vconsole/vconsole-setup.c

```diff
@@ -328,48 +320,88 @@ static void setup_remaining_vcs(int fd, bool utf8) {
         }
 }
 
+static int find_source_vc(char **ret_path, int *ret_idx, const char *src_vc) {
+        int ret_fd;
+
+        if (src_vc) {
+                _cleanup_close_ int fd = -1;
+
+                if (!(*ret_path = malloc(strlen(src_vc) + 1))) {
+                        log_oom();
+                        return -1;
+                }
+                fd = open_terminal(src_vc, O_RDWR|O_CLOEXEC);
+                if (fd < 0) {
+                        log_error_errno(fd, "Failed to open %s: %m", src_vc);
+                        return -errno;
+                }
+
+                if (!is_vconsole(fd)) {
+                        log_error("Device %s is not a virtual console.", src_vc);
+                        return -1;
+                }
+
+                if (!is_allocated_byfd(fd)) {
+                        log_error("Virtual console %s is not allocated.", src_vc);
+                        return -1;
+                }
+
+                if (!is_settable(fd)) {
+                        log_error("Virtual console %s is not in K_XLATE or K_UNICODE.", src_vc);
+                        return -1;
+                }
+
+                strcpy(*ret_path, src_vc);
+                *ret_idx = -1;
+                ret_fd = fd;
+                fd = -1;
+                return ret_fd;
+        }
+
+        if (!(*ret_path = malloc(sizeof("/dev/tty") + DECIMAL_STR_MAX(int) - 2))) {
+                log_oom();
+                return -1;
+        }
+        for (int i = 1; i <= 63; i++) {
+                _cleanup_close_ int fd = -1;
+
+                if (!is_allocated(i))
+                        continue;
+                sprintf(*ret_path, "/dev/tty%i", i);
+                fd = open_terminal(*ret_path, O_RDWR|O_CLOEXEC);
+                if (fd < 0)
+                        continue;
+                if (!is_settable(fd))
+                        continue;
+
+                *ret_idx = i;
+                ret_fd = fd;
+                fd = -1;
+                return ret_fd;
+        }
+
+        log_error("Unable to find a candidate for the source console.");
+        return -1;
```

Comment:
I will return most recent error from the loop iteration now. That should be most useful.

I'll outline or changes in the original thread after the commit.